bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
> To make sure I understand, there would be some notebook issues with this route?

Notebook provides its own version of jquery(ui), so we can expect clashes between both versions. This is similar problem we have with their global `define()`, which interacts with our external resources written in UMD style. We should expect either bokehjs or notebook not working (which is something we went already before).

> Or we'd keep prefixes for the few things that need them?

I don't want to keep prefixes. The whole original point of this PR to get rid of this maintenance nightmare.